dask/dataframe/io.py

```diff
@@ -724,10 +736,59 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
     if (start != 0 or stop is not None) and len(keys) > 1:
         raise NotImplementedError(read_hdf_error_msg)
     from .multi import concat
-    return concat([one_path_one_key(path, k, start, s, columns, chunksize, lock)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize,
+                                    lock, division_column)
                    for k, s in zip(keys, stops)])
 
 
+def _find_divisions(division_column, start, stop, chunksize, path, key, lock):
+    divisions = list()
+    starts = [start]
+    stops = list()
+    row = start
+    if isinstance(division_column, string_types):
+        _div_col = lambda x: x[division_column]
+    elif division_column:
+        _div_col = lambda x: pd.Series(x.index)
+    stop = _get_last_row(path, key, lock, stop)
+    data = _pd_read_hdf(path, key, lock,
+                        {'start': row, 'stop': row + 1})
+    divisions.append(_div_col(data).iloc[0])
+    for row in range(start + chunksize, stop, chunksize):
+        data = _pd_read_hdf(path, key, lock,
+                            {'start': row - 1, 'stop': row + 1})
+        while _div_col(data).iloc[0] == _div_col(data).iloc[1]:
+            row += 1
+            if row >= stop:
+                break 
+            data = _pd_read_hdf(path, key, lock,
+                                {'start': row - 1, 'stop': row + 1})
+        if row >= stop:
+            break
+        starts.append(row)
+        stops.append(row)
+        divisions.append(_div_col(data).iloc[1])
+    stops.append(stop)
+    data = _pd_read_hdf(path, key, lock,
+                        {'start': stop - 1, 'stop': stop})
+    divisions.append(_div_col(data).iloc[0])
+    return divisions, starts, stops
+
+
+def _get_last_row(path, key, lock, stop):
+    if stop is None:
+        if lock:
+            lock.acquire()
+        try:
+            nrows = pd.HDFStore(path).get_storer(key).nrows
+        finally:
+            if lock:
+                lock.release()
+        return nrows
+    else:
+        return stop
```

Comment:
We don't need to deal with locks when making the dataframe.  I suggest removing this function and just using `pd.HDFStore(path).get_storer(key).nrows` inline